torch/_meta_registrations.py

```diff
@@ -214,7 +221,6 @@ def _to_copy(
     device = self.device if device is None else device
     layout = self.layout if layout is None else layout
     assert pin_memory is None
-    assert memory_format is None
     return self.new_empty(self.shape, dtype=dtype, device=device, pin_memory=pin_memory)
```

Comment:
No, you need this assert. Assert it's None or the default contiguity

Reply:
What should I do if it's channels last? Manually create a `new_empty_strided`?

Reply:
If it's channels last then yes, we need to properly respect the requested memory format. It might be easiest to add memory format as an argument to `new_empty`, not sure. Are we using `new_empty` here for composite compliance? `torch.empty` would also let you directly use memory format.

Reply:
Yeah lemme just fix it here: https://github.com/pytorch/pytorch/blob/master/aten/src/ATen/EmptyTensor.cpp#L329